### PR TITLE
Updated Vitamins cap

### PR DESCRIPTION
**Pull request by LOuroboros**
## Description
Starting from Gen. 8, the vitamins no longer have a cap of 100 EVs per stat.
This PR updates that, and it also adds a config for those who would prefer to stick to the old behavior.

## **Discord contact info**
Lunos#4026

### CODE REVIEW

File: include/constants/item_config.h [hunk 1]
```diff
@@ -18,5 +18,6 @@
 #define I_KEY_ESCAPE_ROPE       GEN_7   // In Gen8, Escape Rope became a Key Item. Keep in mind, this will make it free to buy in marts.
 #define I_LEGACY_HEALING_ITEMS  GEN_7   // In Gen7+, certain healing items recover less HP than they used to.
 #define I_SITRUS_BERRY_HEAL     GEN_7   // In Gen4+, Sitrus Berry was changed from healing 30 HP to healing 25% of Max HP.
+#define I_NEW_VITAMIN_EV_CAP    GEN_8   // In Gen8, the Vitamins no longer have a cap of 100 EV per stat.
```
Comment (AsparagusEduardo) [inline include/constants/item_config.h:21]:
```suggestion
#define I_VITAMIN_EV_CAP        GEN_8   // In Gen8, the Vitamins no longer have a cap of 100 EV per stat.
```

File: include/constants/pokemon.h [hunk 2]
```diff
@@ -278,7 +278,11 @@
 #define USE_RANDOM_IVS (MAX_PER_STAT_IVS + 1)
 #define MAX_PER_STAT_EVS 255
 #define MAX_TOTAL_EVS 510
+#if I_NEW_VITAMIN_EV_CAP >= GEN_8
```
Comment (AsparagusEduardo) [inline include/constants/pokemon.h:281]:
```suggestion
#if I_VITAMIN_EV_CAP >= GEN_8
```

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]

**Review by AsparagusEduardo — APPROVED** [review]

File: include/constants/pokemon.h [hunk 3]
```diff
@@ -278,7 +278,11 @@
 #define USE_RANDOM_IVS (MAX_PER_STAT_IVS + 1)
 #define MAX_PER_STAT_EVS 255
 #define MAX_TOTAL_EVS 510
+#if I_VITAMIN_EV_CAP >= GEN_8
+#define EV_ITEM_RAISE_LIMIT MAX_PER_STAT_EVS
+#elif
```
Comment (ghoulslash) [inline include/constants/pokemon.h:283]: this should be `#else`. I'm surprised it compiled

Comment (LOuroboros) [reply include/constants/pokemon.h:283]: Done. My bad.

**Pull request merged**